src/porepy/models/run_convergence_study.py

```diff
@@ -0,0 +1,217 @@
+import os
+import logging
+from typing import (  # noqa
+    Any,
+    Callable,
+    Coroutine,
+    Generator,
+    Generic,
+    Iterable,
+    List,
+    Mapping,
+    Optional,
+    Set,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+)
+from pathlib import Path
+from pprint import pformat
+import functools
+import inspect
+
+import pendulum
+import porepy as pp
+
+from porepy.models.contact_mechanics_model import ContactMechanics
+from porepy.utils.grid_refinement import gb_coarse_fine_cell_mapping, refine_mesh
+from porepy.utils.grid_convergence import grid_error
+
+
+logger = logging.getLogger(__name__)
+
+
+def gb_refinements(
```

Comment:
is this function works also for dfn? i.e., without the higher dimension grid?

Reply:
I would think not - if there are assumptions on the distribution of grids between the dimensions, these should be easy to remove.